src/ws.cpp

```diff
@@ -146,6 +147,7 @@ static const struct wl_surface_interface s_surfaceInterface = {
         auto& surface = *static_cast<Surface*>(wl_resource_get_user_data(surfaceResource));
 
         surface.dmabufBuffer = Instance::singleton().getDmaBufBuffer(bufferResource);
+        surface.shmBuffer = wl_shm_buffer_get(bufferResource);
```

Comment:
Do we want to do this even when surface.dmabufBuffer != nullptr?

Reply:
It avoids verbose branching, and it's pretty cheap since it checks the resource implementation. The two can't both be non-null at the same time because the same resource can't implement two different wl_buffer interfaces.